sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/xml/XmlDataSourceV2.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources.v2.xml
+
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.execution.datasources._
+import org.apache.spark.sql.execution.datasources.v2._
+import org.apache.spark.sql.execution.datasources.xml.XmlFileFormat
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class XmlDataSourceV2 extends FileDataSourceV2 {
```

Comment:
Thanks for submitting this PR. I see that it is essentially a copy of json V2 and refactored for XML. I pointed a few differences in the PR compared to json. Are there any other deviations?

Reply:
Yes, I referred to `JSON` and other other data source implementations that already have v1 and v2. Except for the push down part, the left part should be consistent.